add v2 (python 3.9, 3.10, 3.11)

kidanger opened the pull request:
(images are untested, in particular the gpu ones, they might require higher cuda version on the ipol server)

@qbammey any idea on packages we could add for this version?

qbammey reviewed (commented):
The following packages were required in demos since last May:
- ghostscript
- libopencv-dev
- libboost-all-dev

As for python libraries, I would suggest (both for future use cases and based on what is already required by demos)
- seaborn (current version 0.13.2)
- albumentations (current version 1.4.0)
- networkx (current version 3.2.1)
-  easydict (current version 1.12)
- tifffile (current version 2024.2.12)
- pytorch-lightning (current version 2.2.0)
- magika (current version 0.5.0) could be helpful going forward for type analysis (for video/data input types)
-  sympy (current version 1.12)
- bokeh (current version 3.3.4)
-  altair (current version 5.2.0)

kidanger commented:
> The following packages were required in demos since last May:
> 
>     * ghostscript
> 
>     * libopencv-dev
> 
>     * libboost-all-dev
Added.

> As for python libraries, I would suggest (both for future use cases and based on what is already required by demos)
> 
>     * seaborn (current version 0.13.2)
Users can install it themselves.
>     * albumentations (current version 1.4.0)
It seems to add ~150MB over the current image size. Augmentation is mostly for training, and it's best if the demo does not require training related dependencies.
>     * networkx (current version 3.2.1)
It seems to be quite unrelated to IPOL articles, users can install it themselves.
>     * easydict (current version 1.12)
Small enough that users can install it themselves. (also I would consider this package a bad practice in general)
>     * tifffile (current version 2024.2.12)
Added.
>     * pytorch-lightning (current version 2.2.0)
Added in the pytorch images. (doesn't add much compared to current size)
>     * magika (current version 0.5.0) could be helpful going forward for type analysis (for video/data input types)
I don't think magika is relevant in ipol demo codes.
>     * sympy (current version 1.12)
It seems to be quite unrelated to IPOL articles, users can install it themselves.
>     * bokeh (current version 3.3.4)
Users can install it themselves.
>     * altair (current version 5.2.0)
Users can install it themselves.
